int32 motor
string color    # "black" / "red" / "green" / "blue"